### PR TITLE
Mux modification

### PR DESCRIPTION
**Pull request by elv-reza**
Solves issues: https://github.com/eluv-io/avpipe/issues/67, https://github.com/eluv-io/avpipe/issues/68
- Increase muxing capacity up to 4*4096 ABR segments (more than 8 hours).
- Add capability to mux to MP4 in addition to fMP4. This can be chosen using the format in transcoding parameters.
- During muxing adjust output PTS/DTS and start from zero.
- Better EOF detection when muxing multiple streams.
- Start muxing after the first keyframe detection.
- Some enhancements and fixes in mux output seeker interface.
- Fix a memory leak in avpipe mux when muxing is filished

### CODE REVIEW

File: libavpipe/include/avpipe_xc.h [hunk 1]
```diff
@@ -112,7 +112,7 @@ typedef struct udp_packet_t {
 typedef struct mux_input_ctx_t {
     int     n_parts;                    /* Number of input parts */
     int     index;                      /* Index of current input part that should be processed */
-    char    *parts[MAX_MUX_IN_STREAM];  /* All the input parts */
+    char    **parts;                    /* All the input parts */
```
Comment (elv-serban) [inline libavpipe/include/avpipe_xc.h:115]: Is this better than the array?  Sine we are allocating the entire amount anyway?

**Review by elv-serban — APPROVED** [review]
Looks good

File: libavpipe/src/avpipe_mux.c [hunk 2]
```diff
@@ -228,6 +228,12 @@ avpipe_init_muxer(
     char *out_filename = p->url;
     xctx_t *p_xctx;
 
+    in_mux_ctx->video.parts = (char **) calloc(MAX_MUX_IN_STREAM, sizeof(char *));
```
Comment (elv-nate) [inline libavpipe/src/avpipe_mux.c:231]: Do we know at all how many streams there may be at this point? It seems a little overkill to allocate 16MB of pointers to parts for every single transcode, when the vast majority of the time only a small handful of the streams are used.

**Review by elv-nate — APPROVED** [review]
Looks like it would work no problem.

File: libavpipe/src/avpipe_mux.c (hunk 2 above)
Comment (elv-reza) [reply libavpipe/src/avpipe_mux.c:231]: No, we need to preprocess to pass the number of streams and their index (using some params). This is the most straight forward solution.

File: libavpipe/src/avpipe_xc.c [hunk 3]
```diff
@@ -2257,9 +2257,11 @@ encode_frame(
 
         if (selected_decoded_audio(decoder_context, stream_index) >= 0) {
             /* Set the packet duration if it is not the first audio packet */
-            if (encoder_context->audio_pts[stream_index] != AV_NOPTS_VALUE)
+            if (encoder_context->audio_pts[stream_index] != AV_NOPTS_VALUE) {
                 output_packet->duration = output_packet->pts - encoder_context->audio_pts[stream_index];
-            else
+                if (!strcmp(params->ecodec2, "aac"))
+                    output_packet->duration = 1024;
+            } else
```
Comment (elv-serban) [inline libavpipe/src/avpipe_xc.c:2264]: What does this change address? (It may cover a hole in audio)

**Pull request merged**